src/bloqade/stim/rewrite/squin_noise.py

```diff
@@ -31,6 +34,28 @@ def rewrite_NoiseChannel(
         if rewrite_method is None:
             return RewriteResult()
 
+        if isinstance(stmt, squin_noise.stmts.MultiQubitNoiseChannel):
+            # MultiQubitNoiseChannel represents a broadcast operation, but Stim does not
+            # support broadcasting for multi-qubit noise channels.
+            # Therefore, we must expand the broadcast into individual stim statements.
+            qubit_address_attr = stmt.qubits.hints.get("address", None)
+            if not isinstance(qubit_address_attr, AddressAttribute):
+                return RewriteResult()
+
+            address_tuple = qubit_address_attr.address
+
+            if not isinstance(address_tuple, AddressTuple):
+                return RewriteResult()
+
+            for address in address_tuple.data:
+                qubit_idx_ssas = insert_qubit_idx_from_address(
```

Comment:
Can `qubit_idx_ssas` be `None`? If so, you need to check for that and bail if it is `None`.